string name
int16 type
---
int16 mode
bool success
